test/leo_gateway_web_tests.erl

```diff
@@ -809,6 +810,70 @@ put_object_normal1_([_TermFun, _Node0, Node1]) ->
+    Bin = crypto:rand_bytes(?CHUNKSIZE),
+    {Chunk, Signature} = compute_chunk(PrevSign, SignHead, SignKey, Bin),
+    gen_chunks(Signature, SignHead, SignKey, Remain - ?CHUNKSIZE, <<Acc/binary, Chunk/binary>>).
+
+compute_chunk(PrevSign, SignHead, SignKey, Bin) ->
+    SizeHex = leo_hex:integer_to_hex(byte_size(Bin), 6),
+    ChunkHashBin = crypto:hash(sha256, Bin),
+    BinToSign = <<"AWS4-HMAC-SHA256-PAYLOAD\n",
+                  SignHead/binary,
+                  PrevSign/binary,  "\n",
+                  "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\n",
+                  ChunkHashBin/binary>>,
+    Signature = crypto:hmac(sha256, SignKey, BinToSign),
+    Sign = leo_hex:binary_to_hexbin(Signature),
+    SizeHexBin = list_to_binary(SizeHex),
+    Chunk = <<SizeHexBin/binary, ";", "chunk-signature=", Sign/binary, "\r\n", 
+              Bin/binary, "\r\n">>,
+    {Chunk, Signature}.
+
+put_object_aws_chunked_([_TermFun, _Node0, Node1]) ->
+    fun() ->
+            ok = rpc:call(Node1, meck, new,
+                          [leo_storage_handler_object, [no_link, non_strict]]),
+            ok = rpc:call(Node1, meck, expect,
+                          [leo_storage_handler_object, put, 2, {ok, 1}]),
+            meck:new(leo_s3_auth, [no_link, non_strict]),
+            Signature = <<"642797dcfdf817ac23b553420f52c160847d3747b2e86e5ac9d07cc5e7f60f63">>,
+            SignHead = <<"20150706T051217Z\n20150706/us-east-1/s3/aws4_request\n">>,
+            SignKey = <<"2040321d898be34c82d1db9e132124f11eb18a3d21569d1ed58b460b88954ac7">>,
+            meck:expect(leo_s3_auth, authenticate, 3, {ok, <<"05236">>, {Signature, SignHead, SignKey}}),
+            Chunks = gen_chunks(Signature, SignHead, SignKey, ?AWSCHUNKEDSIZE, <<>>),
+            try
+                {ok, {SC, _Body}} =
+                httpc:request(put, {lists:append(["http://",
+                                                  ?TARGET_HOST,
+                                                  ":8080/testjv4/testFile.large.one"]),
+                                    [{"authorization","AWS4-HMAC-SHA256 Credential=05236/20150706/us-east-1/s3/aws4_request, SignedHeaders=content-length, Signature=642797dcfdf817ac23b553420f52c160847d3747b2e86e5ac9d07cc5e7f60f63"},
+                                     {"x-amz-content-sha256", "STREAMING-AWS4-HMAC-SHA256-PAYLOAD"},
+                                     {"x-amz-decoded-content-length", integer_to_list(?AWSCHUNKEDSIZE)}
+                                    ], "image/png", Chunks},
+                              [], [{full_result, false}]),
+                ?assertEqual(200, SC)
+            catch
+                throw:Reason ->
+                    throw(Reason)
+            after
+                ok = rpc:call(Node1, meck, unload, [leo_storage_handler_object]),
+                ok = meck:unload(leo_s3_auth)
+            end,
+            ok
+    end.
+
 %% proper_([_TermFun, _Node0, _Node1]) ->
```

Comment:
Noted with thanks, I will fix that after completion of `basho_bench_driver` for LeoFS